Example/RxBLE/ViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ViewController.swift
+//  RxBLE
+//
+//  Created by dmsl1805 on 09/25/2017.
+//  Copyright (c) 2017 dmsl1805. All rights reserved.
+//
+
+import UIKit
+
+import RxBLE
+import CoreBluetooth
+import RxSwift
+
+struct BLEConfig {
+    let peripheralIdentifier: UUID
+    let services: [CBUUID]
+    let scanOpptions: [String: Any]
+    let connectOpptions: [String: Any]
+    
+    let peripheralService: CBUUID
+    let peripheralCharacteristic: CBUUID
+}
+
+enum CustomBLEError: Error {
+    case serviceNotFound
+    case characteristicNotFound
+}
+
+class ViewController: UIViewController {
+    let central = CBCentralManager()
+    let disposeBag = DisposeBag()
+    
+    var peripheral: CBPeripheral!
+    
+    let config = BLEConfig(peripheralIdentifier: UUID(uuidString: "Replace with your peripheral identifier")!,
+                           services: ["Replace with your services"].map { CBUUID(string: $0) },
+                           scanOpptions: [:],
+                           connectOpptions: [:],
+                           peripheralService: CBUUID(string: "Replace with your service"),
+                           peripheralCharacteristic: CBUUID(string: "Replace with service's characteristic"))
+    
+    @IBAction func execute(_ sender: Any) {
+        
+        
```

Comment:
replace to more Rx way (combine button tap and central.rx.state). Also, make central.rx.state as BehaviorSubject

Reply:
Done